sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -89,13 +100,95 @@ public override void VisitAttribute(AttributeSyntax node)
         public override void VisitIdentifierName(IdentifierNameSyntax node)
         {
             var symbols = GetSymbols(node, IsKnownIdentifier).ToList();
+
+            TryStoreFieldAccess(node, symbols);
+
             UsedSymbols.UnionWith(symbols);
 
             TryStorePropertyAccess(node, symbols);
 
             base.VisitIdentifierName(node);
         }
 
+        private void TryStoreFieldAccess(IdentifierNameSyntax node, List<ISymbol> symbols)
+        {
+            var access = ParentAccessType(node);
+            if ((access & SymbolAccess.Declaration) != 0)
```

Comment:
Since we make this check multiple times we could extract a method to reuse the code.

Reply:
Something similar with: https://stackoverflow.com/a/417217 might be useful.

Reply:
given we're not using this pattern else in the codebase, I kept it simple